File: processing/src/main/scala/com/datakata/flink/Schemas.scala
package com.datakata.flink

import com.datakata.flink.LineageEmitter.{Dataset, Field}

object Schemas:

  val salesEvent: List[Field] = List(
    Field("saleId", "STRING"),
    Field("salesmanName", "STRING"),
    Field("city", "STRING"),
    Field("country", "STRING"),
    Field("amount", "DOUBLE"),
    Field("product", "STRING"),
    Field("eventTime", "LONG"),
    Field("source", "STRING"),
    Field("ingestionTime", "LONG")
  )

  val topSalesCity: List[Field] = List(
    Field("window_start", "STRING"),
    Field("window_end", "STRING"),
    Field("city", "STRING"),
    Field("total_sales", "DOUBLE"),
    Field("transaction_count", "LONG"),
    Field("rank", "INTEGER")
  )

  val topSalesmanCountry: List[Field] = List(
    Field("window_start", "STRING"),
    Field("window_end", "STRING"),
    Field("salesman_name", "STRING"),
    Field("country", "STRING"),
    Field("total_sales", "DOUBLE"),
    Field("transaction_count", "LONG"),
    Field("rank", "INTEGER")
  )

  // Pre-built dataset references for lineage
  val kafkaPostgres: Dataset = Dataset("kafka", "sales.postgres", salesEvent)
  val kafkaFiles: Dataset = Dataset("kafka", "sales.files", salesEvent)
  val kafkaLegacy: Dataset = Dataset("kafka", "sales.legacy", salesEvent)
  val kafkaUnified: Dataset = Dataset("kafka", "sales.unified", salesEvent)
  val clickhouseTopCity: Dataset = Dataset("clickhouse", "top_sales_city", topSalesCity)
  val clickhouseTopSalesman: Dataset = Dataset("clickhouse", "top_salesman_country", topSalesmanCountry)

File: processing/src/main/java/com/datakata/flink/Schemas.java
package com.datakata.flink;

import com.datakata.flink.LineageEmitter.Dataset;
import com.datakata.flink.LineageEmitter.Field;

import java.util.List;

public final class Schemas {

    private Schemas() {}

    public static final List<Field> SALES_EVENT = List.of(
        new Field("saleId", "STRING"),
        new Field("salesmanName", "STRING"),
        new Field("city", "STRING"),
        new Field("country", "STRING"),
        new Field("amount", "DOUBLE"),
        new Field("product", "STRING"),
        new Field("eventTime", "LONG"),
        new Field("source", "STRING"),
        new Field("ingestionTime", "LONG")
    );

    public static final List<Field> TOP_SALES_CITY = List.of(
        new Field("window_start", "STRING"),
        new Field("window_end", "STRING"),
        new Field("city", "STRING"),
        new Field("total_sales", "DOUBLE"),
        new Field("transaction_count", "LONG"),
        new Field("rank", "INTEGER")
    );

    public static final List<Field> TOP_SALESMAN_COUNTRY = List.of(
        new Field("window_start", "STRING"),
        new Field("window_end", "STRING"),
        new Field("salesman_name", "STRING"),
        new Field("country", "STRING"),
        new Field("total_sales", "DOUBLE"),
        new Field("transaction_count", "LONG"),
        new Field("rank", "INTEGER")
    );

    public static final Dataset KAFKA_POSTGRES = new Dataset("kafka", "sales.postgres", SALES_EVENT);
    public static final Dataset KAFKA_FILES = new Dataset("kafka", "sales.files", SALES_EVENT);
    public static final Dataset KAFKA_LEGACY = new Dataset("kafka", "sales.legacy", SALES_EVENT);
    public static final Dataset KAFKA_UNIFIED = new Dataset("kafka", "sales.unified", SALES_EVENT);
    public static final Dataset CLICKHOUSE_TOP_CITY = new Dataset("clickhouse", "top_sales_city", TOP_SALES_CITY);
    public static final Dataset CLICKHOUSE_TOP_SALESMAN = new Dataset("clickhouse", "top_salesman_country", TOP_SALESMAN_COUNTRY);
}
